src/main/java/io/cos/cas/osf/web/flow/configurer/OsfCasLoginWebflowConfigurer.java

```diff
@@ -360,6 +367,29 @@ private void createOsfInstitutionLoginCheckAction(final Flow flow) {
         );
     }
 
+    /**
+     * Create the unsupported institution login check / preparation action for OSF CAS.
+     *
+     * @param flow the flow
+     */
+    private void createOsfUnsupportedInstitutionLoginCheckAction(final Flow flow) {
+        ActionState action = createActionState(
+                flow,
+                OsfCasWebflowConstants.STATE_ID_OSF_UNSUPPORTED_INSTITUTION_LOGIN_CHECK,
+                OsfCasWebflowConstants.ACTION_ID_OSF_UNSUPPORTED_INSTITUTION_LOGIN_CHECK
+        );
+        createTransitionForState(
+                action,
+                CasWebflowConstants.TRANSITION_ID_ERROR,
+                OsfCasWebflowConstants.VIEW_ID_INSTITUTION_SSO_FAILED
+        );
```

Comment:
Ah, despite my early comments, I realized that the probable reason that you want a middle action is to make sure that: when error happens, we have an "institution login failure" to handle that. However, this does not work as expected.

* Exceptions need to be thrown in order to be caught by the authentication exception handler, which takes care of everything before redirecting the flow to a specific error view. Transit to an error page without going through the handler may lead to unexpected issues.

* The transition I suggested is: go to `VIEW_ID_UNSUPPORTED_INSTITUTION_SSO_INIT` if returned event is `STATE_ID_OSF_UNSUPPORTED_INSTITUTION_LOGIN_CHECK` after `OsfDefaultLoginPreparationAction` is done. There is no authentication error to happen between the transition unless there are programming errors (i.e. wrong ID, incorrect templates, etc.). These errors can't not be caught by the middle action but will trigger the "CAS Unavailable" exception instead.